net-vpn/strongswan/strongswan-5.9.14-r1.ebuild

```diff
@@ -0,0 +1,320 @@
+
+STRONGSWAN_PLUGINS_STD="aes cmac curve25519 des dnskey drbg fips-prf gcm hmac led lookip md5 nonce pem pgp
+pkcs1 pkcs7 pkcs8 pkcs12 pubkey random rc2 revocation sha1 sha2 sshkey systime-fix unity vici x509 xcbc"
+STRONGSWAN_PLUGINS_OPT_DISABLE="kdf"
+STRONGSWAN_PLUGINS_OPT="acert af-alg agent addrblock aesni botan blowfish bypass-lan
+ccm chapoly ctr error-notify forecast files gcm ha ipseckey md4 mgf1 ntru newhope
+openxpki padlock rdrand save-keys sha3 soup test-vectors unbound whitelist xauth-noauth"
+
+for mod in $STRONGSWAN_PLUGINS_STD; do
+	IUSE="${IUSE} +strongswan_plugins_${mod}"
+done
+
+for mod in $STRONGSWAN_PLUGINS_OPT_DISABLE; do
+	IUSE="${IUSE} strongswan_plugins_${mod}"
+done
+
+for mod in $STRONGSWAN_PLUGINS_OPT; do
+	IUSE="${IUSE} strongswan_plugins_${mod}"
+done
+
+COMMON_DEPEND="non-root? (
+		acct-user/ipsec
+		acct-group/ipsec
+	)
+	dev-libs/glib:2
+	gmp? ( >=dev-libs/gmp-4.1.5:= )
+	gcrypt? ( dev-libs/libgcrypt:= )
+	caps? ( sys-libs/libcap )
+	curl? ( net-misc/curl )
+	ldap? ( net-nds/openldap:= )
+	openssl? ( >=dev-libs/openssl-0.9.8:=[-bindist(-)] )
+	mysql? ( dev-db/mysql-connector-c:= )
+	sqlite? ( >=dev-db/sqlite-3.3.1:3 )
+	systemd? ( sys-apps/systemd )
+	networkmanager? ( net-misc/networkmanager )
+	pam? ( sys-libs/pam )
+	strongswan_plugins_unbound? ( net-dns/unbound:= net-libs/ldns:= )"
+
+DEPEND="${COMMON_DEPEND}
+	virtual/linux-sources
+	sys-kernel/linux-headers"
+
+RDEPEND="${COMMON_DEPEND}
+	virtual/logger
+	sys-apps/iproute2
+	!net-vpn/libreswan
+	selinux? ( sec-policy/selinux-ipsec )"
+
+UGID="ipsec"
+
+pkg_setup() {
+	linux-info_pkg_setup
+
+	elog "Linux kernel version: ${KV_FULL}"
+
+	if ! kernel_is -ge 2 6 16; then
+		eerror
+		eerror "This ebuild currently only supports ${PN} with the"
+		eerror "native Linux 2.6 IPsec stack on kernels >= 2.6.16."
+		eerror
+	fi
+
+	if kernel_is -lt 2 6 34; then
+		ewarn
+		ewarn "IMPORTANT KERNEL NOTES: Please read carefully..."
+		ewarn
+
+		if kernel_is -lt 2 6 29; then
+			ewarn "[ < 2.6.29 ] Due to a missing kernel feature, you have to"
+			ewarn "include all required IPv6 modules even if you just intend"
+			ewarn "to run on IPv4 only."
+			ewarn
+			ewarn "This has been fixed with kernels >= 2.6.29."
+			ewarn
+		fi
+
+		if kernel_is -lt 2 6 33; then
+			ewarn "[ < 2.6.33 ] Kernels prior to 2.6.33 include a non-standards"
+			ewarn "compliant implementation for SHA-2 HMAC support in ESP and"
+			ewarn "miss SHA384 and SHA512 HMAC support altogether."
+			ewarn
+			ewarn "If you need any of those features, please use kernel >= 2.6.33."
+			ewarn
+		fi
+
+		if kernel_is -lt 2 6 34; then
+			ewarn "[ < 2.6.34 ] Support for the AES-GMAC authentification-only"
+			ewarn "ESP cipher is only included in kernels >= 2.6.34."
+			ewarn
+			ewarn "If you need it, please use kernel >= 2.6.34."
+			ewarn
+		fi
+	fi
+}
+
+src_configure() {
```

Comment:
some suggestions for `src_configure`. This should be a bit more readable and less error prone. (haven't actually tested it)

```
src_configure() {
	local myeconfargs=(
		--disable-static
		--enable-ikev1
		--enable-ikev2
		--enable-swanctl
		--enable-socket-dynamic
		--enable-cmd
		$(use_enable curl)
		$(use_enable constraints)
		$(use_enable ldap)
		$(use_enable debug leak-detective)
		$(use_enable dhcp)
		$(use_enable eap eap-sim)
		$(use_enable eap eap-sim-file)
		$(use_enable eap eap-simaka-sql)
		$(use_enable eap eap-simaka-pseudonym)
		$(use_enable eap eap-simaka-reauth)
		$(use_enable eap eap-identity)
		$(use_enable eap eap-md5)
		$(use_enable eap eap-aka)
		$(use_enable eap eap-aka-3gpp2)
		$(use_enable eap md4)
		$(use_enable eap eap-mschapv2)
		$(use_enable eap eap-radius)
		$(use_enable eap eap-tls)
		$(use_enable eap eap-ttls)
		$(use_enable eap xauth-eap)
		$(use_enable eap eap-dynamic)
		$(use_enable farp)
		$(use_enable gmp)
		$(use_enable gcrypt)
		$(use_enable mysql)
		$(use_enable networkmanager nm)
		$(use_enable openssl)
		$(use_enable pam xauth-pam)
		$(use_enable pkcs11)
		$(use_enable sqlite)
		$(use_enable systemd)
		$(use_with caps capabilities libcap)
		--with-piddir=/run
		--with-systemdsystemunitdir="$(systemd_get_systemunitdir)"
	)

	if use non-root; then
		myeconfargs+=( --with-user=${UGID} --with-group=${UGID} )
	fi

	# If a user has already enabled db support, those plugins will
	# most likely be desired as well. Besides they don't impose new
	# dependencies and come at no cost (except for space).
	if use mysql || use sqlite; then
		myeconfargs+=( --enable-attr-sql --enable-sql )
	fi

	# strongSwan builds and installs static libs by default which are
	# useless to the user (and to strongSwan for that matter) because no
	# header files or alike get installed... so disabling them is safe.
	if use pam && use eap; then
		myeconfargs+=( --enable-eap-gtc )
	else
		myeconfargs+=( --disable-eap-gtc )
	fi

	for mod in $STRONGSWAN_PLUGINS_STD; do
		if use strongswan_plugins_${mod}; then
			myeconfargs+=( --enable-${mod} )
		fi
	done

	for mod in $STRONGSWAN_PLUGINS_OPT_DISABLE; do
		if ! use strongswan_plugins_${mod}; then
			myeconfargs+=( --disable-${mod} )
		fi
	done

	for mod in $STRONGSWAN_PLUGINS_OPT; do
		if use strongswan_plugins_${mod}; then
			myeconfargs+=( --enable-${mod} )
		fi
	done

	econf "${myeconfargs[@]}"
}
```

Reply:
Thanks for the feedback good point I'll update the ebuild as soon as possible.